### PR TITLE
Add Googletest Framework, CMake setup & CI for IOT Subproject

### PR DESCRIPTION
**Pull request by MisterPeModder**
More info in the README.md of the `IOT` directory.

### CODE REVIEW

File: .github/workflows/build-tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Build-Testing
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (MelvinCou) [inline .github/workflows/build-tests.yml:7]: Trigger workflow only on file changes in `IOT` folder. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

```suggestion
    branches: [ main ]
    paths: [ 'IOT/**' ]
```

File: .github/workflows/build-tests.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
```
Comment (MelvinCou) [inline .github/workflows/build-tests.yml:15]: Run by default in IOT folder. Then you don't need to set working dir for each step

```suggestion
    runs-on: ubuntu-24.04
    defaults:
      run:
        working-directory: IOT
```

File: .github/workflows/build-tests.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Configure CMake
+      working-directory: ./IOT
+      run: cmake -B ${{github.workspace}}/IOT/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DBUILD_TEST_SUITE=TRUE -DENABLE_TEST_COVERAGE=TRUE
+
+    - name: Build
+      working-directory: ./IOT
+      run: cmake --build ${{github.workspace}}/IOT/build --config ${{env.BUILD_TYPE}} -j
```
Comment (MelvinCou) [inline .github/workflows/build-tests.yml:32]: This is not needed for this tiny project, but using all ressources provided by the runner may speed up workflow time ([Github linux runners have 4 cores!](https://docs.github.com/fr/actions/using-jobs/choosing-the-runner-for-a-job#standard-github-hosted-runners-for-public-repositories)).
But we may continue copy paste this file for other projects 👀.

```suggestion
      run: cmake --build ${{github.workspace}}/IOT/build --config ${{env.BUILD_TYPE}} -j$(nproc)
```

**Review by MelvinCou — CHANGES_REQUESTED** [review]
It would be cool to also compile Arduino project in this CI.
Instructions are in the `README.md` file in the `IOT` folder. Or use this: https://github.com/arduino/setup-arduino-cli

**Comment by MisterPeModder** [comment]
> It would be cool to also compile Arduino project in this CI.
> Instructions are in the `README.md` file in the `IOT` folder. Or use this: https://github.com/arduino/setup-arduino-cli

I would have to test that using the real hardware, can I do that in another PR?

**Comment by MelvinCou** [comment]
> > It would be cool to also compile Arduino project in this CI.
> > Instructions are in the `README.md` file in the `IOT` folder. Or use this: https://github.com/arduino/setup-arduino-cli
> 
> I would have to test that using the real hardware, can I do that in another PR?

No need for real hardware, just compile for a target, no upload.

```sh
# you may fill config.h with dummy content, project may not compile without values 
cp main/config.h.sample main/config.h
# update arduino-cli
arduino-cli core update-index
# install Arduino Mega plateform
arduino-cli core install arduino:avr
# compile project, no real Arduino required
arduino-cli compile --fqbn arduino:avr:mega main

# upload to a real Arduino, we don't need this!
# arduino-cli upload -p /dev/ttyACM0 --fqbn arduino:avr:mega main
```

**Review by MelvinCou — APPROVED** [review]

**Pull request merged**